keras/src/backend/openvino/numpy.py

```diff
@@ -505,7 +512,23 @@ def digitize(x, bins):
 
 
 def dot(x, y):
-    raise NotImplementedError("`dot` is not supported with openvino backend")
+    element_type = None
+    if isinstance(x, OpenVINOKerasTensor):
+        element_type = x.output.get_element_type()
+    if isinstance(y, OpenVINOKerasTensor):
+        element_type = y.output.get_element_type()
+    x = get_ov_output(x, element_type)
+    y = get_ov_output(y, element_type)
+    x_shape = ov_opset.shape_of(x)
+    y_shape = ov_opset.shape_of(y)
```

Comment:
you do not need it. Just use `x.get_partial_shape()` before check that rank is static and after retrieve rank value.